File: src/main/java/olivermakesco/de/refmagic/registry/RefinedMagicItems.java
package olivermakesco.de.refmagic.registry;

import net.minecraft.item.BlockItem;
import net.minecraft.item.Item;
import net.minecraft.item.SignItem;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import olivermakesco.de.refmagic.item.*;
import olivermakesco.de.refmagic.Mod;
import org.quiltmc.qsl.item.setting.api.QuiltItemSettings;

import java.sql.Ref;

public class RefinedMagicItems {
    public static Item kyriteShard = new Item(new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item kyritePowder = new Item(new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item kyriteOreEnd = new BlockItem(RefinedMagicBlocks.kyriteOreEnd, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item kyriteOreOverworld = new BlockItem(RefinedMagicBlocks.kyriteOreOverworld, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item kyriteOreNether = new BlockItem(RefinedMagicBlocks.kyriteOreNether, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item kyriteBlock = new BlockItem(RefinedMagicBlocks.kyriteBlock, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item topaz = new Item(new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item topazOre = new BlockItem(RefinedMagicBlocks.topazOre, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item topazBlock = new BlockItem(RefinedMagicBlocks.topazBlock, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item bismuth = new Item(new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item bismuthOreNether = new BlockItem(RefinedMagicBlocks.bismuthOreNether, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item bismuthOreOverworld = new BlockItem(RefinedMagicBlocks.bismuthOreOverworld, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item bismuthBlock = new BlockItem(RefinedMagicBlocks.bismuthBlock, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item citrine = new Item(new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item citrineOre = new BlockItem(RefinedMagicBlocks.citrineOre, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item citrineBlock = new BlockItem(RefinedMagicBlocks.citrineBlock, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item opal = new Item(new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item opalOre = new BlockItem(RefinedMagicBlocks.opalOre, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item opalBlock = new BlockItem(RefinedMagicBlocks.opalBlock, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item roseQuartz = new Item(new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item roseQuartzOre = new BlockItem(RefinedMagicBlocks.roseQuartzOre, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item roseQuartzBlock = new BlockItem(RefinedMagicBlocks.roseQuartzBlock, new Item.Settings().group(RefinedMagicRegistry.group));
    public static Item altarBlock = new BlockItem(RefinedMagicBlocks.altarBlock, new Item.Settings().group(RefinedMagicRegistry.group));

    public static Item dimensionalTotem = new DimensionalTotemItem(new Item.Settings().group(RefinedMagicRegistry.group).maxCount(1));
    public static Item baseNecklace = new NecklaceItem(new Item.Settings().maxCount(1));

    public static Item enchantedStem = new BlockItem(RefinedMagicBlocks.enchantedStem, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item strippedEnchantedStem = new BlockItem(RefinedMagicBlocks.strippedEnchantedStem, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedHyphae = new BlockItem(RefinedMagicBlocks.enchantedHyphae, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item strippedEnchantedHyphae = new BlockItem(RefinedMagicBlocks.strippedEnchantedHyphae, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedPlanks = new BlockItem(RefinedMagicBlocks.enchantedPlanks, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedSlab = new BlockItem(RefinedMagicBlocks.enchantedSlab, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedStairs = new BlockItem(RefinedMagicBlocks.enchantedStairs, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedButton = new BlockItem(RefinedMagicBlocks.enchantedButton, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedPressurePlate = new BlockItem(RefinedMagicBlocks.enchantedPressurePlate, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedTrapdoor = new BlockItem(RefinedMagicBlocks.enchantedTrapdoor, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedDoor = new BlockItem(RefinedMagicBlocks.enchantedDoor, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedFence = new BlockItem(RefinedMagicBlocks.enchantedFence, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedFenceGate = new BlockItem(RefinedMagicBlocks.enchantedFenceGate, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedSign = new SignItem(new QuiltItemSettings().group(RefinedMagicRegistry.group), RefinedMagicBlocks.enchantedSign, RefinedMagicBlocks.enchantedWallSign);

    public static Item enchantedWart = new BlockItem(RefinedMagicBlocks.enchantedWart, new QuiltItemSettings().group(RefinedMagicRegistry.group));

    public static Item enchantedShroomlight = new BlockItem(RefinedMagicBlocks.enchantedShroomlight, new QuiltItemSettings().group(RefinedMagicRegistry.group));

    public static Item enlium = new BlockItem(RefinedMagicBlocks.enlium, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enliumGrowth = new BlockItem(RefinedMagicBlocks.enliumGrowth, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item enchantedFungus = new BlockItem(RefinedMagicBlocks.enchantedFungus, new QuiltItemSettings().group(RefinedMagicRegistry.group));
    public static Item hoopvine = new BlockItem(RefinedMagicBlocks.hoopvine, new QuiltItemSettings().group(RefinedMagicRegistry.group));

    public static Item powerableCopper = new BlockItem(RefinedMagicBlocks.powerableCopper, new QuiltItemSettings().group(RefinedMagicRegistry.group));

    public static void register() {
        register(kyriteShard, Mod.id("kyrite_shard"));
        register(kyritePowder, Mod.id("kyrite_powder"));
        register(kyriteOreOverworld, Mod.id("kyrite_ore_overworld"));
        register(kyriteOreNether, Mod.id("kyrite_ore_nether"));
        register(kyriteOreEnd, Mod.id("kyrite_ore_end"));
        register(kyriteBlock, Mod.id("kyrite_block"));
        register(topaz, Mod.id("topaz"));
        register(topazOre, Mod.id("topaz_ore"));
        register(topazBlock, Mod.id("topaz_block"));
        register(bismuth, Mod.id("bismuth"));
        register(bismuthOreOverworld, Mod.id("bismuth_ore_overworld"));
        register(bismuthOreNether, Mod.id("bismuth_ore_nether"));
        register(bismuthBlock, Mod.id("bismuth_block"));
        register(citrine, Mod.id("citrine"));
        register(citrineOre, Mod.id("citrine_ore"));
        register(citrineBlock, Mod.id("citrine_block"));
        register(opal, Mod.id("opal"));
        register(opalOre, Mod.id("opal_ore"));
        register(opalBlock, Mod.id("opal_block"));
        register(roseQuartz, Mod.id("rose_quartz"));
        register(roseQuartzOre, Mod.id("rose_quartz_ore"));
        register(roseQuartzBlock, Mod.id("rose_quartz_block"));
        register(altarBlock, Mod.id("altar"));

        register(dimensionalTotem, Mod.id("dimensional_totem"));
        register(baseNecklace, Mod.id("necklace"));

        register(enchantedStem, Mod.id("enchanted_stem"));
        register(strippedEnchantedStem, Mod.id("stripped_enchanted_stem"));
        register(enchantedHyphae, Mod.id("enchanted_hyphae"));
        register(strippedEnchantedHyphae, Mod.id("stripped_enchanted_hyphae"));
        register(enchantedPlanks, Mod.id("enchanted_planks"));
        register(enchantedSlab, Mod.id("enchanted_slab"));
        register(enchantedStairs, Mod.id("enchanted_stairs"));
        register(enchantedButton, Mod.id("enchanted_button"));
        register(enchantedPressurePlate, Mod.id("enchanted_pressure_plate"));
        register(enchantedTrapdoor, Mod.id("enchanted_trapdoor"));
        register(enchantedDoor, Mod.id("enchanted_door"));
        register(enchantedFence, Mod.id("enchanted_fence"));
        register(enchantedFenceGate, Mod.id("enchanted_fence_gate"));
        register(enchantedSign, Mod.id("enchanted_sign"));

        register(enchantedShroomlight, Mod.id("enchanted_shroomlight"));

        register(enchantedWart, Mod.id("enchanted_wart"));

        register(enlium, Mod.id("enlium"));
        register(enliumGrowth, Mod.id("enlium_growth"));
        register(enchantedFungus, Mod.id("enchanted_fungus"));
        register(hoopvine, Mod.id("hoopvine"));

        register(powerableCopper, Mod.id("powerable_copper"));
    }

    public static void register(Item item, Identifier id) {
        Registry.register(Registry.ITEM,id,item);
    }
}

File: src/main/kotlin/olivermakesco/de/refmagic/registry/RefinedMagicItems.kt
package olivermakesco.de.refmagic.registry

import net.minecraft.entity.EquipmentSlot
import net.minecraft.item.BlockItem
import net.minecraft.item.Item
import net.minecraft.item.SignItem
import net.minecraft.util.registry.Registry
import olivermakesco.de.refmagic.Mod
import olivermakesco.de.refmagic.item.DimensionalTotemItem
import olivermakesco.de.refmagic.item.NecklaceItem
import olivermakesco.de.refmagic.item.WitheriteArmorItem
import org.quiltmc.qsl.item.setting.api.QuiltItemSettings
import org.quiltmc.qkl.wrapper.minecraft.registry.*

object RefinedMagicItems {
    val baseItemSettings = QuiltItemSettings().group(RefinedMagicRegistry.items)

    @JvmField
    var kyriteShard = Item(baseItemSettings)
    @JvmField
    var kyritePowder = Item(baseItemSettings)
    var kyriteOreEnd: Item = BlockItem(
        RefinedMagicBlocks.kyriteOreEnd,
        baseItemSettings
    )
    var kyriteOreOverworld: Item = BlockItem(
        RefinedMagicBlocks.kyriteOreOverworld,
        baseItemSettings
    )
    var kyriteOreNether: Item = BlockItem(
        RefinedMagicBlocks.kyriteOreNether,
        baseItemSettings
    )
    var kyriteBlock: Item = BlockItem(
        RefinedMagicBlocks.kyriteBlock,
        baseItemSettings
    )
    var topaz = Item(
        baseItemSettings
    )
    var topazOre: Item = BlockItem(
        RefinedMagicBlocks.topazOre,
        baseItemSettings
    )
    var topazBlock: Item = BlockItem(
        RefinedMagicBlocks.topazBlock,
        baseItemSettings
    )
    var bismuth = Item(
        baseItemSettings
    )
    var bismuthOreNether: Item = BlockItem(
        RefinedMagicBlocks.bismuthOreNether,
        baseItemSettings
    )
    var bismuthOreOverworld: Item = BlockItem(
        RefinedMagicBlocks.bismuthOreOverworld,
        baseItemSettings
    )
    var bismuthBlock: Item = BlockItem(
        RefinedMagicBlocks.bismuthBlock,
        baseItemSettings
    )
    var citrine = Item(
        baseItemSettings
    )
    var citrineOre: Item = BlockItem(
        RefinedMagicBlocks.citrineOre,
        baseItemSettings
    )
    var citrineBlock: Item = BlockItem(
        RefinedMagicBlocks.citrineBlock,
        baseItemSettings
    )
    var opal = Item(
        baseItemSettings
    )
    var opalOre: Item = BlockItem(
        RefinedMagicBlocks.opalOre,
        baseItemSettings
    )
    var opalBlock: Item = BlockItem(
        RefinedMagicBlocks.opalBlock,
        baseItemSettings
    )
    var roseQuartz = Item(
        baseItemSettings
    )
    var roseQuartzOre: Item = BlockItem(
        RefinedMagicBlocks.roseQuartzOre,
        baseItemSettings
    )
    var roseQuartzBlock: Item = BlockItem(
        RefinedMagicBlocks.roseQuartzBlock,
        baseItemSettings
    )
    var altarBlock: Item = BlockItem(
        RefinedMagicBlocks.altarBlock,
        baseItemSettings
    )
    var dimensionalTotem: Item = DimensionalTotemItem(
        QuiltItemSettings()
            .group(RefinedMagicRegistry.items)
            .maxCount(1)
    )
    var baseNecklace: Item = NecklaceItem(
        QuiltItemSettings()
            .maxCount(1)
    )
    var enchantedStem: Item = BlockItem(
        RefinedMagicBlocks.enchantedStem,
        baseItemSettings
    )
    var strippedEnchantedStem: Item = BlockItem(
        RefinedMagicBlocks.strippedEnchantedStem,
        baseItemSettings
    )
    var enchantedHyphae: Item = BlockItem(
        RefinedMagicBlocks.enchantedHyphae,
        baseItemSettings
    )
    var strippedEnchantedHyphae: Item = BlockItem(
        RefinedMagicBlocks.strippedEnchantedHyphae,
        baseItemSettings
    )
    var enchantedPlanks: Item = BlockItem(
        RefinedMagicBlocks.enchantedPlanks,
        baseItemSettings
    )
    var enchantedSlab: Item = BlockItem(
        RefinedMagicBlocks.enchantedSlab,
        baseItemSettings
    )
    var enchantedStairs: Item = BlockItem(
        RefinedMagicBlocks.enchantedStairs,
        baseItemSettings
    )
    var enchantedButton: Item = BlockItem(
        RefinedMagicBlocks.enchantedButton,
        baseItemSettings
    )
    var enchantedPressurePlate: Item = BlockItem(
        RefinedMagicBlocks.enchantedPressurePlate,
        baseItemSettings
    )
    var enchantedTrapdoor: Item = BlockItem(
        RefinedMagicBlocks.enchantedTrapdoor,
        baseItemSettings
    )
    var enchantedDoor: Item = BlockItem(
        RefinedMagicBlocks.enchantedDoor,
        baseItemSettings
    )
    var enchantedFence: Item = BlockItem(
        RefinedMagicBlocks.enchantedFence,
        baseItemSettings
    )
    var enchantedFenceGate: Item = BlockItem(
        RefinedMagicBlocks.enchantedFenceGate,
        baseItemSettings
    )
    var enchantedSign: Item = SignItem(
        baseItemSettings,
        RefinedMagicBlocks.enchantedSign,
        RefinedMagicBlocks.enchantedWallSign
    )
    var enchantedWart: Item = BlockItem(RefinedMagicBlocks.enchantedWart,
        baseItemSettings
    )
    var enchantedShroomlight: Item = BlockItem(
        RefinedMagicBlocks.enchantedShroomlight,
        baseItemSettings
    )
    var enlium: Item = BlockItem(
        RefinedMagicBlocks.enlium,
        baseItemSettings
    )
    var enliumGrowth: Item = BlockItem(
        RefinedMagicBlocks.enliumGrowth,
        baseItemSettings
    )
    var enchantedFungus: Item = BlockItem(
        RefinedMagicBlocks.enchantedFungus,
        baseItemSettings
    )
    var hoopvine: Item = BlockItem(
        RefinedMagicBlocks.hoopvine,
        baseItemSettings
    )
    var witheredHeart = Item(
        QuiltItemSettings().
        group(RefinedMagicRegistry.items)
    )
    var witheriteBlob = Item(
        baseItemSettings
    )
    var infestedNetherrackBlock: Item = BlockItem(
        RefinedMagicBlocks.infestedNetherrack,
        baseItemSettings
    )
    var witheriteHelmet: Item = WitheriteArmorItem(
            EquipmentSlot.HEAD,
            baseItemSettings
        )
    var witheriteChestplate: Item = WitheriteArmorItem(
        EquipmentSlot.CHEST,
        baseItemSettings
    )
    var witheriteLeggings: Item = WitheriteArmorItem(
        EquipmentSlot.LEGS,
        baseItemSettings
    )
    var witheriteBoots: Item = WitheriteArmorItem(
        EquipmentSlot.FEET,
        baseItemSettings
    )
    fun register() {
        Registry.ITEM(Mod.modid) {
            kyriteShard             withName "kyrite_shard"
            topaz                   withName "topaz"
            bismuth                 withName "bismuth"
            citrine                 withName "citrine"
            opal                    withName "opal"
            roseQuartz              withName "rose_quartz"
            kyritePowder            withName "kyrite_powder"
            kyriteOreOverworld      withName "kyrite_ore_overworld"
            kyriteOreNether         withName "kyrite_ore_nether"
            kyriteOreEnd            withName "kyrite_ore_end"
            topazOre                withName "topaz_ore"
            bismuthOreOverworld     withName "bismuth_ore_overworld"
            bismuthOreNether        withName "bismuth_ore_nether"
            citrineOre              withName "citrine_ore"
            opalOre                 withName "opal_ore"
            roseQuartzOre           withName "rose_quartz_ore"
            kyriteBlock             withName "kyrite_block"
            topazBlock              withName "topaz_block"
            bismuthBlock            withName "bismuth_block"
            citrineBlock            withName "citrine_block"
            opalBlock               withName "opal_block"
            roseQuartzBlock         withName "rose_quartz_block"
            altarBlock              withName "altar"
            dimensionalTotem        withName "dimensional_totem"
            witheredHeart           withName "withered_heart"
            witheriteBlob           withName "witherite_blob"
            infestedNetherrackBlock withName "infested_netherrack"
            witheriteHelmet         withName "witherite_helmet"
            witheriteChestplate     withName "witherite_chestplate"
            witheriteLeggings       withName "witherite_leggings"
            witheriteBoots          withName "witherite_boots"
            baseNecklace            withName "necklace"
            enchantedStem           withName "enchanted_stem"
            strippedEnchantedStem   withName "stripped_enchanted_stem"
            enchantedHyphae         withName "enchanted_hyphae"
            strippedEnchantedHyphae withName "stripped_enchanted_hyphae"
            enchantedPlanks         withName "enchanted_planks"
            enchantedSlab           withName "enchanted_slab"
            enchantedStairs         withName "enchanted_stairs"
            enchantedButton         withName "enchanted_button"
            enchantedPressurePlate  withName "enchanted_pressure_plate"
            enchantedTrapdoor       withName "enchanted_trapdoor"
            enchantedDoor           withName "enchanted_door"
            enchantedFence          withName "enchanted_fence"
            enchantedFenceGate      withName "enchanted_fence_gate"
            enchantedSign           withName "enchanted_sign"
            enchantedShroomlight    withName "enchanted_shroomlight"
            enchantedWart           withName "enchanted_wart"
            enlium                  withName "enlium"
            enliumGrowth            withName "enlium_growth"
            enchantedFungus         withName "enchanted_fungus"
            hoopvine                withName "hoopvine"
        }
    }
}